packages/foreman/foreman/foreman.spec

```diff
@@ -909,18 +864,18 @@ mkdir db/ \\
 cp -rf %{_datadir}/%{name}/db/* db/ \\
 mv db/schema.rb.nulldb db/schema.rb \\
 \\
-ln -s %{?scl_prefix_nodejs:%{_scl_root}}%{nodejs_sitelib} node_modules \\
-export GEM_PATH=%%{buildroot}%%{gem_dir}:\${GEM_PATH:+\${GEM_PATH}}\${GEM_PATH:-\`%{?scl:scl enable %%{scl} -- }ruby -e "print Gem.path.join(':')"\`} \\
+ln -s %{nodejs_sitelib} node_modules \\
+export GEM_PATH=%%{buildroot}%%{gem_dir}:\${GEM_PATH:+\${GEM_PATH}}\${GEM_PATH:-\`ruby -e "print Gem.path.join(':')"\`} \\
 unlink tmp \\
 \\
 rm \`pwd\`/config/initializers/encryption_key.rb \\
 rm \`pwd\`/config/database.yml \\
-/usr/bin/%%{?scl:%%{scl}-}rake security:generate_encryption_key \\
+%{rake} security:generate_encryption_key \\
 export BUNDLER_EXT_NOSTRICT=1 \\
 export NODE_ENV=production \\
 cp %%{buildroot}%%{%{name}_bundlerd_dir}/%%{gem_name}.rb ./bundler.d/%%{gem_name}.rb \\
-%%{?-s:/usr/bin/%%{?scl:%%{scl}-}rake %%{-r*}%%{!?-r:plugin:assets:precompile[%%{-n*}%%{!?-n:%%{gem_name}}]} RAILS_ENV=production DATABASE_URL=nulldb://nohost --trace} \\
-%%{?-a:/usr/bin/%%{?scl:%%{scl}-}rake plugin:apipie:cache[%%{gem_name}] FOREMAN_APIPIE_LANGS=en_US RAILS_ENV=production cache_part=resources OUT=%%{buildroot}%%{%{name}_apipie_cache_plugin} DATABASE_URL=nulldb://nohost --trace} \\
+%%{?-s:%{rake} %%{-r*}%%{!?-r:plugin:assets:precompile[%%{-n*}%%{!?-n:%%{gem_name}}]} RAILS_ENV=production DATABASE_URL=nulldb://nohost --trace} \\
+%%{?-a:%{rake} plugin:apipie:cache[%%{gem_name}] FOREMAN_APIPIE_LANGS=en_US RAILS_ENV=production cache_part=resources OUT=%%{buildroot}%%{%{name}_apipie_cache_plugin} DATABASE_URL=nulldb://nohost --trace} \\
```

Comment:
Please check this carefully. I think this results in an explicit `/usr/bin/rake` in the generated macro

Reply:
I see in the generated file:

```
%{?-s:/usr/bin/rake %{-r*}%{!?-r:plugin:assets:precompile[%{-n*}%{!?-n:%{gem_name}}]} RAILS_ENV=production DATABASE_URL=nulldb://nohost --trace} \
%{?-a:/usr/bin/rake plugin:apipie:cache[%{gem_name}] FOREMAN_APIPIE_LANGS=en_US RAILS_ENV=production cache_part=resources OUT=%{buildroot}%{foreman_apipie_cache_plugin} DATABASE_URL=nulldb://nohost --trace} \
```

which looks correct to me?

Reply:
It does, but macros in macros is always tricky. Always good to be careful around that.